gossip/privdata/pvtdataprovider.go

```diff
@@ -202,27 +202,44 @@ func (pdp *PvtdataProvider) RetrievePvtdata(pvtdataToRetrieve []*ledger.TxPvtdat
 
 	pvtdata := make(rwsetByKeys)
 
+	//If there is no private data to retrieve for the block, skip all population attempts and return
+	if len(pvtdataRetrievalInfo.eligibleMissingKeys) == 0 {
+		pdp.logger.Debugf("No eligible collection private write sets to fetch for block [%d]", pdp.blockNum)
+		retrievedPvtdata.pvtdataRetrievalInfo = pvtdataRetrievalInfo
```

Comment:
Nit. Not really needed. For the code consistency, I am fine with having this line.

Reply:
It's still necessary to set the `retrievedPvtdata.pvtdataRetrievalInfo` for purging later